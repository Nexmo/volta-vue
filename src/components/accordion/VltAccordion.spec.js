import { mount } from '@vue/test-utils';
import { VltAccordionItem, VltAccordion } from '.';

describe('vlt-accordion', () => {
  it('should be rendered', () => {
    const wrapper = mount(VltAccordion);
    wrapper.setProps({ id: 'accq', classNames: 'accordion' });
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('vlt-accordion-item', () => {
  it('should be without the active class', () => {
    const wrapper = mount(VltAccordionItem);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should be with the active class', () => {
    const wrapper = mount(VltAccordionItem);
    wrapper.setProps({ expanded: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should be with the active class after trigger is clicked', () => {
    const wrapper = mount(VltAccordionItem);
    wrapper.find('.Vlt-accordion__trigger').trigger('click');
    expect(wrapper.element).toMatchSnapshot();
  });
});
