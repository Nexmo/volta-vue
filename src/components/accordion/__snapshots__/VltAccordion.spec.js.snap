// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-accordion should be rendered 1`] = `
<div
  class="accordion Vlt-accordion"
  id="accq"
>
  
  
</div>
`;

exports[`vlt-accordion-item should be with the active class 1`] = `
<div
  class="Vlt-accordion__item_opened Vlt-accordion__item"
>
  <div
    class="Vlt-accordion__trigger_active Vlt-accordion__trigger"
  >
    
    <div>
      Need header content
    </div>
    
  </div>
  <div
    class="Vlt-accordion__content_open Vlt-accordion__content"
  >
    
    
  </div>
</div>
`;

exports[`vlt-accordion-item should be with the active class after trigger is clicked 1`] = `
<div
  class="Vlt-accordion__item_opened Vlt-accordion__item"
>
  <div
    class="Vlt-accordion__trigger_active Vlt-accordion__trigger"
  >
    
    <div>
      Need header content
    </div>
    
  </div>
  <div
    class="Vlt-accordion__content_open Vlt-accordion__content"
  >
    
    
  </div>
</div>
`;

exports[`vlt-accordion-item should be without the active class 1`] = `
<div
  class="Vlt-accordion__item"
>
  <div
    class="Vlt-accordion__trigger"
  >
    
    <div>
      Need header content
    </div>
    
  </div>
  <div
    class="Vlt-accordion__content"
  >
    
    
  </div>
</div>
`;
